python/ray/tune/suggest/optuna.py

```diff
@@ -261,33 +293,56 @@ def __init__(self,
                 "You passed an initialized sampler to `OptunaSearch`. The "
                 "`seed` parameter has to be passed to the sampler directly "
                 "and will be ignored.")
+        elif sampler:
+            assert isinstance(
+                self._sampler,
+                BaseSampler), ("You can only pass an instance of "
+                               "`optuna.samplers.BaseSampler` "
+                               "as a sampler to `OptunaSearcher`.")
 
-        self._sampler = sampler or ot.samplers.TPESampler(seed=seed)
-
-        assert isinstance(self._sampler, BaseSampler), \
-            "You can only pass an instance of `optuna.samplers.BaseSampler` " \
-            "as a sampler to `OptunaSearcher`."
+        self._sampler = sampler
+        self._seed = seed
 
         self._ot_trials = {}
         self._ot_study = None
         if self._space:
             self._setup_study(mode)
 
-    def _setup_study(self, mode: str):
+    def _setup_study(self, mode: Union[str, list]):
         if self._metric is None and self._mode:
+            if isinstance(self._mode, list):
+                raise ValueError(
+                    "If ``mode`` is a list (multi-objective optimization "
+                    "case), ``metric`` must be defined.")
             # If only a mode was passed, use anonymous metric
             self._metric = DEFAULT_METRIC
 
         pruner = ot.pruners.NopPruner()
         storage = ot.storages.InMemoryStorage()
 
+        if self._sampler:
+            sampler = self._sampler
+        elif isinstance(mode, list):
+            sampler = ot.samplers.MOTPESampler(seed=self._seed)
+        else:
+            sampler = ot.samplers.TPESampler(seed=self._seed)
```

Comment:
```suggestion
    else
            sampler = ot.samplers.TPESampler(seed=self._seed)
```

`MOTPESampler` is now deprecated and `TPESampler` works even if the objective has multiple values. 😸 

https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.MOTPESampler.html#optuna.samplers.MOTPESampler

Reply:
Oh, that's great! I will leave MOTPE as we want to support older versions of Optuna too, but use TPE with the newer versions.